pkg/storage/cacher.go

```diff
@@ -389,7 +399,7 @@ func newCacherListerWatcher(storage Interface, resourcePrefix string, newListFun
 // Implements cache.ListerWatcher interface.
 func (lw *cacherListerWatcher) List() (runtime.Object, error) {
 	list := lw.newListFunc()
-	if err := lw.storage.List(context.TODO(), lw.resourcePrefix, 0, Everything, list); err != nil {
+	if err := lw.storage.List(context.TODO(), lw.resourcePrefix, "", Everything, list); err != nil {
```

Comment:
Actually no.
The "cacherListerWatcher" is the class that is reponsible for updating the internal cacher based on etcd state. So what it is doing it has an open "watch" for all resources of a given type to etcd and all "real watchers" are just accessing cache.

The List() here is used to either initiate the first watch or to relist in case e.g. falling out of etcd window.
This means that there will never be a corresponding user request here.

Why do we change it to ""? It's because we would like to force the list from etcd - if you pass an empty string, this will cause exactly that.
So this is WAI.